pkg/runctl/run_manager.go

```diff
@@ -381,7 +401,37 @@ func (c *runManager) setupNetworkPolicyFromConfig(runNamespace string) error {
 	return nil
 }
 
-func (c *runManager) createTektonTaskRun(pipelineRun k8s.PipelineRun) error {
+func (c *runManager) volumesWithSaSecret(ctx *runContext) []corev1api.Volume {
+	var mode int32 = 420
+	return []corev1api.Volume{
+		corev1api.Volume{Name: "service-account-token",
+			VolumeSource: corev1api.VolumeSource{
+				Secret: &corev1api.SecretVolumeSource{
+					SecretName:  c.getServiceAccountSecretName(ctx),
+					DefaultMode: &mode,
+				},
+			},
+		}}
+}
+
+func (c *runManager) getServiceAccountSecretName(ctx *runContext) string {
+	if c.testing != nil {
+		if c.testing.getServiceAccountSecretNameStub != nil {
+			return c.testing.getServiceAccountSecretNameStub(ctx)
+		}
+		return ""
+	}
+	if repeatGetServiceAccountSecretName {
+		return ctx.serviceAccount.GetServiceAccountSecretNameRepeat()
+	}
+	result := ctx.serviceAccount.GetServiceAccountSecretName()
+	if result == "" {
+		panic("service account secret not found")
+	}
```

Comment:
This is test code and this does not belong here.

Reply:
Removed